### PR TITLE
gomodguard: fix problem where duplicate issues were being reported

### PR DESCRIPTION
**Pull request by ryancurrah**
Gomodguard was saving issues to a global variable and when it was the process issues functioned returned results it would end up returning duplicates. This issue solves that by the process function returning only new issues found.

Also ensure we only configure and initialize gomodguard once by removing those steps from the `Run` func.

https://github.com/ryancurrah/gomodguard/compare/v1.2.0...v1.2.1

### CODE REVIEW

File: pkg/golinters/gomodguard.go [hunk 1]
```diff
@@ -21,75 +24,70 @@ func NewGomodguard() *goanalysis.Linter {
 		issues   []goanalysis.Issue
 		mu       = sync.Mutex{}
 		analyzer = &analysis.Analyzer{
-			Name: goanalysis.TheOnlyAnalyzerName,
-			Doc:  goanalysis.TheOnlyanalyzerDoc,
+			Name: gomodguardName,
+			Doc:  gomodguardDesc,
```
Comment (ldez) [inline pkg/golinters/gomodguard.go:28]: Could revert this change?
The `TheOnlyAnalyzer` is a specific pattern.

**Review by ldez — CHANGES_REQUESTED** [review]

File: pkg/golinters/gomodguard.go (hunk 1 above)
Comment (ryancurrah) [reply pkg/golinters/gomodguard.go:28]: Yep.

Comment (ryancurrah) [reply pkg/golinters/gomodguard.go:28]: Both or just the name?

Comment (ldez) [reply pkg/golinters/gomodguard.go:28]: you can keep `goanalysis.TheOnlyanalyzerDoc` only

Comment (ryancurrah) [reply pkg/golinters/gomodguard.go:28]: Reverted both.

File: pkg/golinters/gomodguard.go [hunk 2]
```diff
@@ -28,68 +31,63 @@ func NewGomodguard() *goanalysis.Linter {
...
+		}
 
-			processor, err := gomodguard.NewProcessor(processorCfg)
-			if err != nil {
-				lintCtx.Log.Warnf("running gomodguard failed: %s: if you are not using go modules "+
-					"it is suggested to disable this linter", err)
-				return nil, nil
-			}
+		analyzer.Run = func(pass *analysis.Pass) (interface{}, error) {
+			files := []string{}
```
Comment (ldez) [inline pkg/golinters/gomodguard.go:75]:
```suggestion
			var files []string
```

**Review by ldez — CHANGES_REQUESTED** [review]
nitpic

**Comment by ryancurrah** [comment]
> nitpic

resolved

**Review by ldez — APPROVED** [review]
LGTM
